### DISCUSSION TITLE
lossless webp and webp quality 'Save Media As' option

### DISCUSSION THREAD

**Discussion by BPplays** [Kareadita/Kavita#3039] [Ideas]
### Idea Description

the current webp option uses lossy mode with no quality option, i would like an option for lossless webp and an option to control lossy quality would be nice too.
on [the file i tested](https://raw.githubusercontent.com/BPplays/images/main/kavita_sug_manga_cover_example_test/v42747_c47215.png) i measured the file size as png: 1938 KB. and as lossless webp: 1038 KB.
i used cwebp with `-lossless -m 6`

for lossy:
quality 100 
　\- 647 KB

quality 95   
　\- 482 KB

quality 90  
　\- 370 KB

quality 75   
　\- 204 KB

### Idea Category

Feature Enhancement

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by Cducharme84** [comment]
Don't forget to include the cwebp result for lossy default equivalent to your command  (-m 6 being a change of how long each image is processed for not a measure of quality I followed that for the lossy): 199KB, I suspect since we're talking covers for cards rather than the images that should be full screen high-resolution in a reader most folks won't want a 5X file size ballooning, so I'm not sure of the utility as cards should load faster than a full sized page image when off-network. Just a heads up if you don't wind up with expected traction on this.

That said, I'm all for options so you've my upvote! :)

**Reply by BPplays** [reply]
yeah i know -m changes the processing time and -q changes lossy quality, when i made this i mostly wanted a lossless webp option because its a lot smaller then png so it should save disk space and load a bit faster even with no quality loss

also what do you mean by cards? this post is about `Admin Dashboard > Media > Save Media As` which says 'All media Kavita manages (covers, bookmarks, favicons) will be encoded as this type.'

**Reply by Cducharme84** [reply]
So, I forgot the bookmark function which _is_ a full-screen function that I forget because I don't use so good catch there! The reference to cards is a reference to how the covers are typically displayed. 

When looking at a series/collection view the cover and name/library pairing you see in the UI are referred to as "cards" or "list", essentially though covers/favicons aren't displaying at the full screen so the majority of users would be likely to not favor lossless when comparing file size and UX effect were it limited to solely those two things.

My summation of that changes though with bookmarks, people that do like that feature would definitely benefit from a lossless variant of the webp conversion as it would represent space savings over png with retention of quality in that full screen display.

**Reply by BPplays** [reply]
@Cducharme84 oh, i guess i forgot to specify exactly why i want lossless.
in mihon if you use the kavita extension you can import manga, and you can view covers in fullscreeen if you tap them and even normally when you're in the sub menu for a manga the cover is pretty big (in tablet mode) and i think i tested a while ago and could notice a difference between Cover Image Size large and extra large like that

**Reply by Cducharme84** [reply]
Ohh it's so easy to forget folks using apps when you're as satisfied reading on mobile with the PWA mode as I am. I can see how that'd be useful for you guys and hope it becomes an option!

Interesting how the -q 100 for lossy was still lower than the lossless in file size, I have had such different results than that in most of my tests. But of course, image conversion is such a crapshoot.
